src/perfoutputwidgetkonsole.cpp

```diff
@@ -0,0 +1,166 @@
+/*
+  perfoutputwidgetkonsole.cpp
+
+  This file is part of Hotspot, the Qt GUI for performance analysis.
+
+  Copyright (C) 2021 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Lieven Hey <lieven.hey@kdab.com>
+
+  Licensees holding valid commercial KDAB Hotspot licenses may use this file in
+  accordance with Hotspot Commercial License Agreement provided with the Software.
+
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "perfoutputwidgetkonsole.h"
+
+#include <QEvent>
+#include <QKeyEvent>
+#include <QStandardPaths>
+#include <QTemporaryFile>
+#include <QVBoxLayout>
+
+#include <KParts/ReadOnlyPart>
+#include <KParts/kde_terminal_interface.h>
+#include <KService>
+
+PerfOutputWidgetKonsole::PerfOutputWidgetKonsole(QWidget* parent)
+    : PerfOutputWidget(parent)
+{
+    auto layout = new QVBoxLayout(this);
+    setLayout(layout);
+}
+
+PerfOutputWidgetKonsole::~PerfOutputWidgetKonsole() = default;
+
+PerfOutputWidgetKonsole* PerfOutputWidgetKonsole::create(QWidget* parent)
+{
+    auto part = createPart();
+    if (!part) {
+        return nullptr;
+    }
+
+    delete part;
```

Comment:
huh why is the part deleted here again? shouldn't it be passed on to the console widget?

also you should check that the part actually implements the terminal interface too

finally I guess you should only check findTail once here, and then set it as a member arg of the widget and pass it along through the ctor too?

i.e. one way or another, more of the `::clear` code below should be extracted and then called here to check that all dependencies are available before creating the widget